File: 알고리즘/유찬홍/10866.c
#include <stdio.h>
#include <string.h>

int front = 10000, rear = 10000;

int is_empty() {
    if (front == rear) return 1;
    else return 0;
}

void push_front(int *deque, int a) {
    deque[front--] = a;
}

void push_back(int *deque, int a) {
    deque[++rear] = a;
}

int pop_front(int *deque) {
    if (is_empty()) return -1;
    return deque[++front];
}

int pop_back(int *deque) {
    if (is_empty()) return -1;
    return deque[rear--];
}

int main() {
    int n, a, deque[20002];
    char arr[15];
    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        scanf("%s", arr);
        if (!strcmp(arr, "push_front")) {
            scanf("%d", &a);
            push_front(deque, a);
        } else if (!strcmp(arr, "push_back")) {
            scanf("%d", &a);
            push_back(deque, a);
        } else if (!strcmp(arr, "pop_front")) printf("%d\n", pop_front(deque));
        else if (!strcmp(arr, "pop_back")) printf("%d\n", pop_back(deque));
        else if (!strcmp(arr, "size")) printf("%d\n", rear - front);
        else if (!strcmp(arr, "empty")) printf("%d\n", is_empty());
        else if (!strcmp(arr, "front")) printf("%d\n", is_empty() ? -1 : deque[front + 1]);
        else if (!strcmp(arr, "back")) printf("%d\n", is_empty() ? -1 : deque[rear]);
    }
    return 0;
}

File: 알고리즘/정지관/javascript/10866.js
let fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().split("\n");
let queue = [];
let answer = [];

for (let i = 1; i <= parseInt(input[0]); i++) {
  let arr = input[i].split(" ");
  switch (arr[0]) {
    case "push_front":
      queue.unshift(arr[1]);
      break;
    case "push_back":
      queue.push(arr[1]);
      break;
    case "pop_front":
      if (queue.length === 0) answer.push(-1);
      else answer.push(queue.shift());
      break;
    case "pop_back":
      if (queue.length === 0) answer.push(-1);
      else answer.push(queue.pop());
      break;
    case "size":
      answer.push(queue.length);
      break;
    case "empty":
      if (queue.length === 0) {
        answer.push(1);
      } else {
        answer.push(0);
      }
      break;
    case "front":
      if (queue.length === 0) answer.push(-1);
      else answer.push(queue[0]);
      break;
    case "back":
      if (queue.length === 0) answer.push(-1);
      else answer.push(queue[queue.length - 1]);
      break;
  }
}

console.log(answer.join("\n"));
